### PR TITLE
Make it possible to delegate the timer handling to the backend

### PR DESCRIPTION
**Pull request by tronical**
When building with no_std, the backend can seed the Instant wrapper and
provide updates to it.

### CODE REVIEW

File: sixtyfps_runtime/corelib/animations.rs [hunk 1]
```diff
@@ -68,30 +68,54 @@ impl core::ops::SubAssign<core::time::Duration> for Instant {
...
+    }
+
+    fn duration_since_start() -> core::time::Duration {
+        #[cfg(feature = "std")]
+        {
+            let the_beginning = *INITIAL_INSTANT.get_or_init(|| instant::Instant::now());
+            return instant::Instant::now() - the_beginning;
+        }
+        #[cfg(not(feature = "std"))]
+        crate::backend::instance().unwrap().duration_since_start()
```
Comment (ogoffart) [inline sixtyfps_runtime/corelib/animations.rs:96]: why not always delegate to the backend, but the default implementation would use instand::Instant

**Review by ogoffart — COMMENTED** [review]
the adventage of having the backend always tell the time is that the testing backend can implement its own time passing time so that we can have test for the timer code (which currently is not tested,  only the animation are tested)

In a way, it would be nice if the timer and the animation used the same time reference.

File: sixtyfps_runtime/corelib/animations.rs (hunk 1 above)
Comment (tronical) [reply sixtyfps_runtime/corelib/animations.rs:96]: The main issue with that is that for the desktop builds we may not always have a backend yet. The API crate initializes it but we use timers early on in main.

Comment (ogoffart) [reply sixtyfps_runtime/corelib/animations.rs:92]: and then we don't even need to store initial_instant in this crate.

Comment (tronical) [reply sixtyfps_runtime/corelib/animations.rs:92]: Ok :)

**Review by ogoffart — APPROVED** [review]

**Pull request merged**